aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -89,21 +97,97 @@ pub fn pop_value_by_type(ty_arg: &Type, args: &mut VecDeque<Value>) -> SafeNativ
 pub fn create_value_by_type(ty_arg: &Type, value: u128) -> SafeNativeResult<Value> {
     match ty_arg {
         Type::U128 => Ok(Value::u128(value)),
-        Type::U64 => {
-            if let Ok(cast_value) = u64::try_from(value) {
-                Ok(Value::u64(cast_value))
-            } else {
-                Err(SafeNativeError::InvariantViolation(PartialVMError::new(
-                    StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR,
-                )))
-            }
-        },
+        Type::U64 => Ok(Value::u64(u128_to_u64(value)?)),
         _ => Err(SafeNativeError::Abort {
             abort_code: EUNSUPPORTED_AGGREGATOR_TYPE,
         }),
     }
 }
 
+// To aviod checking is_string_type multiple times, check type_arg only once, and convert into this enum
+enum SnapshotType {
+    U128,
+    U64,
+    String,
+}
+
+impl SnapshotType {
+    fn from_ty_args(context: &SafeNativeContext, ty_arg: &Type) -> SafeNativeResult<Self> {
+        match ty_arg {
+            Type::U128 => Ok(Self::U128),
+            Type::U64 => Ok(Self::U64),
+            _ => {
+                // Check if the type is a string
+                if is_string_type(context, ty_arg)? {
+                    Ok(Self::String)
+                } else {
+                    // If not a string, return an error
+                    Err(SafeNativeError::Abort {
+                        abort_code: EUNSUPPORTED_AGGREGATOR_SNAPSHOT_TYPE,
+                    })
+                }
+            },
+        }
+    }
+
+    pub fn pop_snapshot_field_by_type(
+        &self,
+        args: &mut VecDeque<Value>,
+    ) -> SafeNativeResult<SnapshotValue> {
+        self.parse_snapshot_value_by_type(aggregator_snapshot_field_value(&safely_pop_arg!(
```

Comment:
Just trying to learn the design patterns. Is there generally any benefit of using a lot of such small 1-3 line functions in the code? I heard that it's generally preferrable to have a decent amount of stuff going on each function.

Reply:
when same code is used in the multiple places, and name of the function adds to the explanation of what the short code does - it helps a lot in readability.

generally, things that are repeated, is useful to extract and avoid repetition